### DISCUSSION TITLE
how to update only one or part of the nodes in the graph

### DISCUSSION THREAD

**Discussion by hellomuyi** [pyg-team/pytorch_geometric#4743] [Q&A]
Hello, I would like to ask, how to update only one or part of the nodes in the graph in PyG's GAT. To be precise, I only need to get the attention weight of a node or some nodes (not all nodes) in the graph to its neighbors, I read the source code, that is, "torch_geometric.utils.softmax", the bottom layer uses operations such as torch.scatter_, but I still can't solve it.

If no such method exists,  is there an efficient way to extract the attention weights of **partial** center nodes to their neighbors from all the attention weights (alpha: (num_edges, heads))?

**Comment by rusty1s** [comment]
You should be able to only pass in a single node and its neighbors to `GATConv`:
```python
edge_index = torch.tensor([[1, 2, 3, 4, 5], [0, 0, 0, 0, 0]])  # bipartite graph only pointing to node 0
x = ...  # node feature matrix with shape [6, #features]

out_central = conv((x, x[:1]), edge_index)
# out_central has shape [1, #features]
```
Let me know if this resolves your issues :)

**Reply by hellomuyi** [reply]
Thank you. That's all right. But I'm a little confused about this example. I think node 0 should be updated like this:
```py
edge_index = torch.tensor([[0, 1, 2, 3, 4], [0, 0, 0, 0, 0]])  # bipartite graph only pointing to node 0
x = ...  # node feature matrix with shape [6, #features]

out_central = conv((x[1:], x[:1]), edge_index)
# out_central has shape [1, #features]
```
or
```py
edge_index = torch.tensor([[1, 2, 3, 4, 5], [0, 0, 0, 0, 0]])  # bipartite graph only pointing to node 0
x = ...  # node feature matrix with shape [6, #features]

out_central = conv((x, x[:1]), edge_index)
# out_central has shape [1, #features]
```
Did I misunderstand something?

**Reply by rusty1s** [reply]
No, you are correct. I fixed it in my response as well.
